### PR TITLE
[TTS] Add volume passthrough to fp for riva

### PR DESCRIPTION
**Pull request by blisc**
# What does this PR do ?

Add volume passthrough to fp for riva

**Collection**: TTS

# Changelog 
- Add volume passthrough to fp for riva

**PR Type**:
- [x] New Feature

### CODE REVIEW

File: nemo/collections/tts/modules/fastpitch.py [hunk 1]
```diff
@@ -298,8 +298,18 @@ def infer(self, *, text, pitch=None, speaker=None, pace=1.0):
...
         dec_out, _ = self.decoder(input=len_regulated, seq_lens=dec_lens)
         spect = self.proj(dec_out).transpose(1, 2)
-        return spect.to(torch.float), dec_lens, durs_predicted, log_durs_predicted, pitch_predicted
+        return (
+            spect.to(torch.float),
+            dec_lens,
+            durs_predicted,
+            log_durs_predicted,
+            pitch_predicted,
+            volume_extended.squeeze(-1).float(),
```
Comment (redoctopus) [inline nemo/collections/tts/modules/fastpitch.py:313]: Won't this error if `volume` is None?

File: nemo/collections/tts/models/fastpitch.py [hunk 2]
```diff
@@ -520,6 +520,7 @@ def _prepare_for_export(self, **kwargs):
             "text": NeuralType(('B', 'T_text'), TokenIndex()),
             "pitch": NeuralType(('B', 'T_text'), RegressionValuesType()),
             "pace": NeuralType(('B', 'T_text'), optional=True),
+            "volume": NeuralType(('B', 'T_text')),
```
Comment (itzsimpl) [inline nemo/collections/tts/models/fastpitch.py:523]: shouldn't this be optional to not break old models?

File: nemo/collections/tts/models/fastpitch.py [hunk 3]
```diff
@@ -572,5 +575,5 @@ def input_example(self, max_batch=1, max_dim=256):
 
         return (inputs,)
 
-    def forward_for_export(self, text, pitch, pace, speaker=None):
-        return self.fastpitch.infer(text=text, pitch=pitch, pace=pace, speaker=speaker)
+    def forward_for_export(self, text, pitch, pace, volume, speaker=None):
```
Comment (itzsimpl) [inline nemo/collections/tts/models/fastpitch.py:578]: shouldn't this be `volume=None`?

File: nemo/collections/tts/modules/fastpitch.py (hunk 1 above)
Comment (blisc) [reply nemo/collections/tts/modules/fastpitch.py]: Updated

File: nemo/collections/tts/models/fastpitch.py (hunk 2 above)
Comment (blisc) [reply nemo/collections/tts/models/fastpitch.py:523]: It's not a model feature. It wouldn't affect other model versions.

**Review by redoctopus — DISMISSED** [review]

**Review by redoctopus — APPROVED** [review]

**Pull request merged**